controllers/src/main/java/com/redhat/ipaas/controllers/integration/IntegrationController.java

```diff
@@ -74,13 +72,17 @@ public void start() {
             if (event!=null && "change-event".equals(event)) {
                 try {
                     ChangeEvent changeEvent = Json.mapper().readValue(data, ChangeEvent.class);
-                    if ((changeEvent != null
-                        || changeEvent.getKind().isPresent() && changeEvent.getId().isPresent())
-                        && (Kind.from(changeEvent.getKind().get()) == Kind.Integration)) {
-                        enqueueCheckIntegration(changeEvent.getId().get());
+                    if (changeEvent != null) {
+                        changeEvent.getId().ifPresent(
+                            id -> changeEvent.getKind().ifPresent(
+                                kind -> {
+                                    if (Kind.from(kind) == Kind.Integration) {
+                                        enqueueCheckIntegration(id);
```

Comment:
@chirino I think its false to enqueue a check for each change on the integration as during processing the event the integration gets updated as well (like status change or adding the GitHub repo). So you end up with multiple jobs for the same integration which then all start a build.

See e.g this logs.

<details>

```
2017-04-19 08:28:14.564  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.i.IntegrationController          : Integration 3 is not in the desired status Activated (current: Draft) --> enqueuing
2017-04-19 08:28:14.566  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.i.IntegrationController          : Processing integration 3.
2017-04-19 08:28:15.346  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.p.c.DefaultProjectGenerator        : Adding step null [integration Optional[3]]
2017-04-19 08:28:15.347  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.catalog.ConnectorCatalog         : Downloading Maven GAV: com.redhat.ipaas:twitter-mention-connector:0.3.6
2017-04-19 08:28:48.919  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.p.c.DefaultProjectGenerator        : Adding step null [integration Optional[3]]
2017-04-19 08:28:48.919  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.p.c.DefaultProjectGenerator        : Adding step null [integration Optional[3]]
2017-04-19 08:28:48.921  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.catalog.ConnectorCatalog         : Downloading Maven GAV: com.redhat.ipaas:salesforce-upsert-contact-connector:0.3.6
2017-04-19 08:29:00.679  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.i.IntegrationController          : Integration 3 is not in the desired status Activated (current: Pending) --> enqueuing
2017-04-19 08:29:00.679  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.i.IntegrationController          : Integration 3 is not in the desired status Activated (current: Pending) --> enqueuing
2017-04-19 08:30:00.680  INFO [-,,,] 1 --- [pool-5-thread-1] c.r.i.c.i.IntegrationController          : Integration 3 is not in the desired status Activated (current: Pending) --> enqueuing
2017-04-19 08:30:00.681  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.i.IntegrationController          : Processing integration 3.
2017-04-19 08:30:01.377  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.p.c.DefaultProjectGenerator        : Adding step null [integration Optional[3]]
2017-04-19 08:30:01.377  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.catalog.ConnectorCatalog         : Downloading Maven GAV: com.redhat.ipaas:twitter-mention-connector:0.3.6
2017-04-19 08:30:01.388  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.p.c.DefaultProjectGenerator        : Adding step null [integration Optional[3]]
2017-04-19 08:30:01.388  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.p.c.DefaultProjectGenerator        : Adding step null [integration Optional[3]]
2017-04-19 08:30:01.388  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.catalog.ConnectorCatalog         : Downloading Maven GAV: com.redhat.ipaas:salesforce-upsert-contact-connector:0.3.6
2017-04-19 08:30:12.274  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.i.IntegrationController          : Integration 3 is not in the desired status Activated (current: Pending) --> enqueuing
2017-04-19 08:30:12.274  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.i.IntegrationController          : Integration 3 is not in the desired status Activated (current: Pending) --> enqueuing
2017-04-19 08:31:12.274  INFO [-,,,] 1 --- [pool-5-thread-1] c.r.i.c.i.IntegrationController          : Integration 3 is not in the desired status Activated (current: Pending) --> enqueuing
2017-04-19 08:31:12.275  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.i.IntegrationController          : Processing integration 3.
2017-04-19 08:31:12.919  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.p.c.DefaultProjectGenerator        : Adding step null [integration Optional[3]]
2017-04-19 08:31:12.919  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.catalog.ConnectorCatalog         : Downloading Maven GAV: com.redhat.ipaas:twitter-mention-connector:0.3.6
2017-04-19 08:31:12.930  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.p.c.DefaultProjectGenerator        : Adding step null [integration Optional[3]]
2017-04-19 08:31:12.930  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.p.c.DefaultProjectGenerator        : Adding step null [integration Optional[3]]
2017-04-19 08:31:12.930  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.catalog.ConnectorCatalog         : Downloading Maven GAV: com.redhat.ipaas:salesforce-upsert-contact-connector:0.3.6
2017-04-19 08:31:22.462  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.i.IntegrationController          : Integration 3 is not in the desired status Activated (current: Pending) --> enqueuing
2017-04-19 08:31:22.463  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.i.IntegrationController          : Integration 3 is not in the desired status Activated (current: Pending) --> enqueuing
2017-04-19 08:32:22.463  INFO [-,,,] 1 --- [pool-5-thread-1] c.r.i.c.i.IntegrationController          : Integration 3 is not in the desired status Activated (current: Pending) --> enqueuing
2017-04-19 08:32:22.464  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.i.IntegrationController          : Processing integration 3.
2017-04-19 08:32:23.103  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.p.c.DefaultProjectGenerator        : Adding step null [integration Optional[3]]
2017-04-19 08:32:23.103  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.catalog.ConnectorCatalog         : Downloading Maven GAV: com.redhat.ipaas:twitter-mention-connector:0.3.6
2017-04-19 08:32:23.113  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.p.c.DefaultProjectGenerator        : Adding step null [integration Optional[3]]
2017-04-19 08:32:23.113  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.p.c.DefaultProjectGenerator        : Adding step null [integration Optional[3]]
2017-04-19 08:32:23.114  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.catalog.ConnectorCatalog         : Downloading Maven GAV: com.redhat.ipaas:salesforce-upsert-contact-connector:0.3.6
2017-04-19 08:32:32.362  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.i.IntegrationController          : Integration 3 is not in the desired status Activated (current: Pending) --> enqueuing
2017-04-19 08:32:32.363  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.i.IntegrationController          : Integration 3 is not in the desired status Activated (current: Pending) --> enqueuing
2017-04-19 08:33:32.363  INFO [-,,,] 1 --- [pool-5-thread-1] c.r.i.c.i.IntegrationController          : Integration 3 is not in the desired status Activated (current: Pending) --> enqueuing
2017-04-19 08:33:32.364  INFO [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.i.IntegrationController          : Processing integration 3.
2017-04-19 08:33:32.403 ERROR [-,,,] 1 --- [pool-4-thread-1] c.r.i.c.i.IntegrationController          : Error while processing integration status for integration 3

```
<details>

Any idea how we should resolved this without introducing a central object holding all actively running checks ? // @jimmidyson